pulsar4s-cats-effect/src/main/scala/com/sksamuel/pulsar4s/cats/CatsAsyncHandler.scala

```diff
@@ -13,8 +13,8 @@ import scala.util.{Failure, Success, Try}
 
 class CatsAsyncHandler extends AsyncHandler[IO] {
 
-  implicit def completableVoidToIO(f: CompletableFuture[Void]): IO[Unit] = completableToIO(f).map(_ => ())
-  implicit def completableToIO[T](f: CompletableFuture[T]): IO[T] =
+  implicit def completableVoidToIO(f: => CompletableFuture[Void]): IO[Unit] = completableToIO(f).map(_ => ())
+  implicit def completableToIO[T](f: => CompletableFuture[T]): IO[T] =
```

Comment:
@sksamuel I think it makes sense to make this lazy, since `IO` and `Task` are meant to be lazy, and we don't want the operation to actually be performed until evaluation time. This is also why I created a `negativeAcknowledgeAsync` method here even though the actual operation is not asynchronous.